pkg/statsd/aggregator.go

```diff
@@ -171,6 +171,10 @@ func (a *MetricAggregator) Process(f ProcessFunc) {
 }
 
 func (a *MetricAggregator) isExpired(now, ts gostatsd.Nanotime) bool {
+	if a.expiryInterval == 0 {
```

Comment:
This effectively pulls a portion of the below check out.  I believe that's a good thing because it's clearer, but it should also be removed from the below check as well.